### DISCUSSION TITLE
Network Quadlet not generating the service?

### DISCUSSION THREAD

**Discussion by haleksandre** [containers/podman#19838] [Q&A] [answered]
In my butane file I have something like so:

```yml
variant: fcos
version: 1.5.0
passwd:
  users:
    - name: core
      groups:
        - sudo
      # ... ssh keys
storage:
  files:
    - path: /etc/containers/systemd/server.network
      contents:
        inline: |
          [Network]
          Subnet=192.168.1.0/24  
          Gateway=192.168.1.1
          NetworkName=server
          Label=lan.media.network=server
    # ... other network, volume
    - path: /etc/containers/systemd/nginx.container
      contents:
        inline: |
          [Unit]
          Description=Nginx Quadlet
          Requires=podman.socket
          After=podman.socket

          [Container]
          Image=docker.io/nginxproxy/nginx-proxy:alpine
          ContainerName=nginx
          AutoUpdate=registry
          Environment=DOCKER_HOST=unix://${XDG_RUNTIME_DIR}/podman/podman.sock
          Network=server.network
          PublishPort=80:80/tcp
          PublishPort=443:443/tcp
          Volume=${XDG_RUNTIME_DIR}/podman/podman.sock:/tmp/docker.sock:ro
          Volume=/var/srv/nginx/latest/conf.d/:/etc/nginx/conf.d/:z 
          Volume=/var/srv/nginx/latest/global/:/etc/nginx/global/:z 
          Volume=/var/srv/nginx/latest/ssl/:/etc/nginx/ssl/:z 

          [Service]
          User=core
          Group=core
          Restart=always
          TimeoutStartSec=900

          [Install]
          WantedBy=multi-user.targe
      # ... other containers
```
but every container fails as the networks & volumes aren't getting created, on first boot, & are getting a `server-network.service not found` error. Am I missing something somewhere to make the Quadlet configuration work properly?

**Comment by ygalblum** [comment]
The `Network` keyword in a `.container` file does not create the network. It instructs Qualdet to add `--network=<network name>` to the `podman run` command. 
So, you are missing the `.network` file that defines the network and instructs Quadlet to create a service to create the network
You can see an example in [this](https://github.com/ygalblum/quadlet-demo) repository:
1. `.container` file: https://github.com/ygalblum/quadlet-demo/blob/main/quadlet-files/quadlet-demo-mysql.container
2. `network` file: https://github.com/ygalblum/quadlet-demo/blob/main/quadlet-files/quadlet-demo.network (All the keys are actually optional)

**Reply by haleksandre** [reply]
Doesn't the first file `server.network` in my butane config create the network that the container will use?

Edit:
I should have mentioned that I'm trying to spin up a Fedore CoreOS machine. This butane config initializes the CoreOS on first boot with certain configurations/files. I'm creating a container using the example from here: https://docs.fedoraproject.org/en-US/fedora-coreos/running-containers/ . The problem seems that the `.network` & `.volume` files don't get picked up for some reason? Perhaps there are written in the wrong directory? On second thought maybe this has more to do with how Fedora CoreOS than Podman.

**Reply by ygalblum** [reply, marked as answer]
Oh, sorry. My eyes went strait to the `.container` file and missed the `.network` one. 

I think the issue is with the `NetworkName` key in the `.network` file. It was only recently added, so, I'm not sure the version you are using supports it. You can check `journalctl` for errors or do a dry-run by running: `/usr/libexec/podman/quadlet --dryrun`.

Please note that if you're using this network only via other Quadlet files, it is not really needed

**Reply by ygalblum** [reply]
Mid-air collision, the comment about `NetworkName` is relevant also for `VolumeName`

**Reply by haleksandre** [reply]
I've looked at the logs & spotted this was the issue, `NetworkName` & `VolumeName` were erroring my quadlet files. Thank you!
